### PR TITLE
AST-1213 - test case for button font styling under the customizer

### PR DESCRIPTION
**Pull request by bsfsnehal**
### Description

Created test case for :
1. button font family
2.  button font size
3.  button font weight
4. button text transform
5.  line height
6. letter spacing
### Screenshots

https://share.getcloudapp.com/YEuPAmr6
### How has this been tested?

Execution: npm run test:e2e:interactive — tests/e2e/specs/customizer/global/button/button-font.test.js
### Checklist:
- [ ] My code is tested
- [ ] My code passes the PHPCS tests
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [ ] I've added proper labels to this pull request

### CODE REVIEW

File: tests/e2e/specs/customizer/global/button/button-font.test.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
+import { insertBlock, createURL, createNewPost, publishPost } from '@wordpress/e2e-test-utils';
```
Comment (bsfsnehal) [inline tests/e2e/specs/customizer/global/button/button-font.test.js:1]: need to change path for publish-post

File: tests/e2e/specs/customizer/global/button/button-font.test.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+import { setCustomize } from '../../../../utils/customize';
+import { setBrowserViewport } from '../../../../utils/set-browser-viewport';
+import { responsiveFontSize } from '../../../../utils/responsive-utils';
+describe( 'Global button font setting under the Customizer', () => {
+	it( 'button font should apply correctly', async () => {
+		const btnFont = {
```
Comment (bsfsnehal) [inline tests/e2e/specs/customizer/global/button/button-font.test.js:7]:
```suggestion
		const buttonFont = {
```

File: tests/e2e/specs/customizer/global/button/button-font.test.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+		await expect( {
+			selector: '.wp-block-button__link',
+			property: 'font-size',
+		} ).cssValueToBe( `${ buttonFontSize[ 'font-size-button' ].desktop }${ buttonFontSize[ 'font-size-button' ][ 'desktop-unit' ] }` );
+
+		await setBrowserViewport( 'medium' );
+		await expect( {
+			selector: '.wp-block-button__link',
+			property: 'font-size',
+		} ).cssValueToBe(
```
Comment (GauravKhupse) [inline tests/e2e/specs/customizer/global/button/button-font.test.js:54]: @bsfsnehal Error on PR please check

Comment (bsfsnehal) [reply tests/e2e/specs/customizer/global/button/button-font.test.js:54]: Please take latest pull.

Comment (bsfsnehal) [reply tests/e2e/specs/customizer/global/button/button-font.test.js]: As discussed, I have commented the code for responsive mode.

File: tests/e2e/specs/customizer/global/button/button-font.test.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+			await page.keyboard.type( 'font' );
+			ppStatus = await publishPost();
+		}
+		await page.goto( createURL( '/buttonFontFamily/' ), {
+			waitUntil: 'networkidle0',
+		} );
+		await page.waitForSelector( '.wp-block-button .wp-block-button__link, input#submit, button, form[CLASS*="wp-block-search__"].wp-block-search .wp-block-search__inside-wrapper .wp-block-search__button, input[type="submit"]' );
+		await expect( {
+			selector: '.wp-block-button .wp-block-button__link, input#submit, button, form[CLASS*="wp-block-search__"].wp-block-search .wp-block-search__inside-wrapper .wp-block-search__button, input[type="submit"]',
+			property: 'font-family',
```
Comment (SimranChougule1) [inline tests/e2e/specs/customizer/global/button/button-font.test.js:23]: add Selector for comment button option

File: tests/e2e/specs/customizer/global/button/button-font.test.js [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+		} ).cssValueToBe( `${ buttonFont[ 'font-family-button' ] }` );
+	} );
```
[the remark this answers is not in the document]
Comment (bsfsnehal) [reply tests/e2e/specs/customizer/global/button/button-font.test.js:44]: updated

File: tests/e2e/specs/customizer/global/button/button-font.test.js (hunk 4 above)
Comment (bsfsnehal) [reply tests/e2e/specs/customizer/global/button/button-font.test.js:38]: Added

**Review by GauravKhupse — APPROVED** [review]

**Review by SimranChougule1 — APPROVED** [review]

**Pull request merged**